goal_src/levels/common/battlecontroller.gc

```diff
@@ -5,3 +5,1043 @@
+   (set! (-> v1-2 user-int8 0) 1)
+   )
+  0
+  (none)
+  )
+
+(defstate battlecontroller-die (battlecontroller)
+  :virtual #t
+  :event
+  (the-as
+   (function process int symbol event-message-block object :behavior battlecontroller)
+   process-drawable-fuel-cell-handler
+   )
+  :code
+  (behavior ()
+   (sound-play-by-name
+    (static-sound-name "money-pickup")
+    (new-sound-id)
+    1024
+    -2286
+    0
+    (the-as uint 1)
+    (the-as vector #f)
+    )
+   (battlecontroller-battle-end)
+   (battlecontroller-camera-off)
+   (battlecontroller-set-task-completed)
+   (countdown (gp-1 (entity-actor-count (-> self entity) 'fade-actor))
+    (let ((s5-1 (new 'stack-no-clear 'event-message-block)))
+     (set! (-> s5-1 from) self)
+     (set! (-> s5-1 num-params) 0)
+     (set! (-> s5-1 message) 'fade)
+     (let ((s4-0 send-event-function)
+           (v1-2 (entity-actor-lookup (-> self entity) 'fade-actor gp-1))
+           )
+      (s4-0 (if v1-2
+             (-> v1-2 extra process)
+             )
+       s5-1
+       )
+      )
+     )
+    )
+   (countdown (gp-2 (entity-actor-count (-> self entity) 'alt-actor))
+    (let ((s5-2 (new 'stack-no-clear 'event-message-block)))
+     (set! (-> s5-2 from) self)
+     (set! (-> s5-2 num-params) 0)
+     (set! (-> s5-2 message) 'trigger)
+     (let ((s4-1 send-event-function)
+           (v1-8 (entity-actor-lookup (-> self entity) 'alt-actor gp-2))
+           )
+      (s4-1 (if v1-8
+             (-> v1-8 extra process)
+             )
+       s5-2
+       )
+      )
+     )
+    )
+   (when
+    (and
+     (battlecontroller-special-contents?)
+     (not (battlecontroller-special-contents-collected?))
+     )
+    (label cfg-15)
+    (birth-pickup-at-point
+     (-> self final-pickup-spawn-point)
+     (-> self final-pickup-type)
+     (the float (-> self entity extra perm task))
+     #f
+     self
+     #f
+     )
+    (when (not (-> self child))
+     (suspend)
+     (goto cfg-15)
+     )
+    (while (-> self child)
+     (suspend)
+     )
+    (battlecontroller-set-special-contents-collected)
+    )
+   (while (-> self child)
+    (suspend)
+    )
+   (process-entity-status! self (entity-perm-status dead) #t)
+   (none)
+   )
+  :post
+  (the-as (function none :behavior battlecontroller) #f)
+  )
+
+(defmethod relocate battlecontroller ((obj battlecontroller) (arg0 int))
+  (dotimes (v1-0 (-> obj spawner-count))
+   (let ((a0-3 (-> obj spawner-array v1-0)))
+    (if (nonzero? (-> a0-3 path))
+     (&+! (-> a0-3 path) arg0)
+     )
+    )
+   )
+  (if (nonzero? (-> obj path-spawn))
+   (&+! (-> obj path-spawn) arg0)
+   )
+  (the-as
+   battlecontroller
+   ((the-as
+     (function process-drawable int process-drawable)
+     (find-parent-method battlecontroller 7)
+     )
+    obj
+    arg0
+    )
+   )
+  )
+
+(defmethod deactivate battlecontroller ((obj battlecontroller))
+  (with-pp
+   (let ((gp-0 pp))
+    (let ((s6-1 obj))
```

Comment:
this is changing the process pointer, which is something we don't handle correctly in the decompiler yet.  can you leave a note at the top of the file about this? https://github.com/water111/jak-project/issues/882